File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/viewModels/ShearingLogVM.ts
import { timestampToDate } from "../utils/helpers";

// Клас для відображення записів стрижки
export class ShearingLogVM {
    id: number; // Ідентифікатор запису стрижки
    date: string; // Дата стрижки
    woolAmount: number; // Кількість вовни в кг
    shepherdName: string | null; // Ім'я пастуха
    shepherdSurname: string | null; // Прізвище пастуха
    sheepId: number; // Ідентифікатор вівці

    constructor(id: number, timestamp: number, woolAmount: number, sheepId: number, shepherdName: string | null, shepherdSurname: string | null) {
        this.id = id;
        this.woolAmount = woolAmount / 1000;
        this.shepherdName = shepherdName;
        this.shepherdSurname = shepherdSurname;
        this.sheepId = sheepId;
        this.date = timestampToDate(timestamp);
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/view_models/shearing_log_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

// Структура, що представляє представлення логу стрижки (ViewModel)
#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct ShearingLogVM {
    id: u64, // Ідентифікатор логу стрижки
    timestamp: u64, // Час стрижки (у вигляді таймстемпу)
    wool_amount: u64, // Кількість вовни
    shepherd_name: Option<String>, // Ім'я пастуха (може бути відсутнім)
    shepherd_surname: Option<String>, // Прізвище пастуха (може бути відсутнім)
    sheep_id: u64 // Ідентифікатор вівці
}
